evennia/contrib/turnbattle/tb_items.py

```diff
@@ -0,0 +1,1374 @@
+
+    #
+    # any commands you add below will overload the default ones.
+    #
+    self.add(tb_items.BattleCmdSet())
+
+This module is meant to be heavily expanded on, so you may want to copy it
+to your game's 'world' folder and modify it there rather than importing it
+in your game and using it as-is.
+"""
+
+from random import randint
+from evennia import DefaultCharacter, Command, default_cmds, DefaultScript
+from evennia.commands.default.muxcommand import MuxCommand
+from evennia.commands.default.help import CmdHelp
+from evennia.utils.spawner import spawn
+from evennia import TICKER_HANDLER as tickerhandler
+
+"""
+----------------------------------------------------------------------------
+OPTIONS
+----------------------------------------------------------------------------
+"""
+
+TURN_TIMEOUT = 30 # Time before turns automatically end, in seconds
+ACTIONS_PER_TURN = 1 # Number of actions allowed per turn
+NONCOMBAT_TURN_TIME = 30 # Time per turn count out of combat
+
+"""
+----------------------------------------------------------------------------
+COMBAT FUNCTIONS START HERE
+----------------------------------------------------------------------------
+"""
+
+def roll_init(character):
+    """
+    Rolls a number between 1-1000 to determine initiative.
+
+    Args:
+        character (obj): The character to determine initiative for
+
+    Returns:
+        initiative (int): The character's place in initiative - higher
+        numbers go first.
+
+    Notes:
+        By default, does not reference the character and simply returns
+        a random integer from 1 to 1000.
+
+        Since the character is passed to this function, you can easily reference
+        a character's stats to determine an initiative roll - for example, if your
+        character has a 'dexterity' attribute, you can use it to give that character
+        an advantage in turn order, like so:
+
+        return (randint(1,20)) + character.db.dexterity
+
+        This way, characters with a higher dexterity will go first more often.
+    """
+    return randint(1, 1000)
+
+
+def get_attack(attacker, defender):
+    """
+    Returns a value for an attack roll.
+
+    Args:
+        attacker (obj): Character doing the attacking
+        defender (obj): Character being attacked
+
+    Returns:
+        attack_value (int): Attack roll value, compared against a defense value
+            to determine whether an attack hits or misses.
+
+    Notes:
+        This is where conditions affecting attack rolls are applied, as well.
+        Accuracy Up and Accuracy Down are also accounted for in itemfunc_attack(),
+        so that attack items' accuracy is affected as well.
+    """
+    # For this example, just return a random integer up to 100.
+    attack_value = randint(1, 100)
+    # Add 25 to the roll if the attacker has the "Accuracy Up" condition.
+    if "Accuracy Up" in attacker.db.conditions:
+        attack_value += 25
```

Comment:
I know these values are here partly as examples, but these kind of numbers seem to be perfect to have up top as `ACCURACY_UP_STEP` etc. Not only does it move the data values to one place where they are easy to change, it also serves as a good example for people expanding on the system (the more these numbers are spread out in the module the easier it is to miss updating one when tweaking). 

This should be considered for the other cases below too, I won't point those out explicitly.

Reply:
That's a good point! I'll add condition values to the options.